src/components/Input/__test__/Input.test.tsx

```diff
@@ -107,4 +109,54 @@ describe("Input", () => {
         expect(input).not.toHaveClass("deriv-input--error");
         expect(inputLabel).toHaveClass("deriv-input__label--disabled");
     });
+
+    it("should update character counter as user types", async () => {
+        const handleChange = jest.fn();
+        const maxLength = 10;
+        render(
+            <Input
+                label="Username"
+                shouldShowCounter
+                maxLength={maxLength}
+                onChange={handleChange}
+            />
+        );
+        const input = screen.getByLabelText("Username");
+        await userEvent.type(input, "input")
+        expect(handleChange).toHaveBeenCalledTimes(5);
```

Comment:
```suggestion
        const input = screen.getByLabelText("Username");
        await userEvent.type(input, "input")
        expect(handleChange).toHaveBeenCalledTimes(5);
        const counterElement = screen.getByText('5/10');
        expect(counterElement).toBeInTheDocument();
```

I think we should check this as well. because the handler can be triggered but doesn't change the counter